bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3402,6 +3403,8 @@ void ReplicaImp::onSeqNumIsStable(SeqNum newStableSeqNum, bool hasStateInformati
     IControlHandler::instance()->onStableCheckpoint();
     ControlStateManager::instance().wedge();
   }
+  auto ret =
+      std::async(std::launch::async, [this, newStableSeqNum]() { createDbCheckpointCb_.invokeAll(newStableSeqNum); });
```

Comment:
the async way of an invocation is a callback implementation details and should be hidden from a caller.

Reply:
fixed